src/main/java/org/terasology/projectile/ProjectileAuthoritySystem.java

```diff
@@ -127,8 +127,8 @@ public void update(float delta) {
             ProjectileActionComponent projectile = entity.getComponent(ProjectileActionComponent.class);
             ProjectileMotionComponent projectileMotion = entity.getComponent(ProjectileMotionComponent.class);
 
-            if(projectileMotion.distanceTravelled >= projectile.maxDistance && projectile.maxDistance != -1) {
-                if(projectile.reusable)
+            if (projectileMotion.distanceTravelled >= projectile.maxDistance && projectile.maxDistance != -1) {
+                if (projectile.reusable)
                     entity.send(new DeactivateProjectileEvent());
                 else
                     entity.destroy();
```

Comment:
@portokaliu Look at the code above. If the `distanceTravelled` is greater than `maxDistance`, the changed part below would never be reached.
There is a `continue;` just below this line.